client/src/hooks/datafiles/mutations/toolbarAppUtils.ts

```diff
@@ -72,6 +72,9 @@ export function getAllocationForToolbarAction(
     activeAllocations.length > 0 &&
     appObj
   ) {
+    if (!appObj.execSystems[0].canRunBatch) {
+      return 'VM'; // if app runs on VM, no allocation needed, so bypass allocation check
+    }
```

Comment:
Would it be better to do a `find` on this with the `appObj.definition.execSystemId`?

Reply:
Great idea! i was trying to think of an alternative since we have the dynamic exec system possibility now, I'll implement this now